core/trino-main/src/main/java/io/trino/Session.java

```diff
@@ -348,7 +348,7 @@ public Session beginTransactionId(TransactionId transactionId, TransactionManage
             String catalogName = entry.getKey();
             SelectedRole role = entry.getValue();
             if (transactionManager.getCatalogHandle(transactionId, catalogName).isEmpty()) {
-                throw new TrinoException(CATALOG_NOT_FOUND, "Catalog for role not found: " + catalogName);
+                throw new TrinoException(CATALOG_NOT_FOUND, "Catalog '%s' for role not found".formatted(catalogName));
```

Comment:
Can we reword? What is this supposed to mean "Catalog for role?"

Reply:
I think it's the roles that have access to the catalog? Also as a side note, the naming of `catalogRoles`/`connectorRoles` is not consistent in the `io.trino.spi.security.Identity` class

Reply:
The whole catalog vs connector thing is inconsistent.

For the error message we have to figure out the details. Please let me know here and I can try to come up with a message text.

Reply:
I  would just go with

Catalog not found


since we dont say what role anyway and depending on the access control there might be other things at play